File: nearest_neighbor/nn.c
// nevesuser
#include <stdio.h>
int main(void)
{
    double points[][3] = {{1, 2, 3}, {3, 4, 5}, {5, 6, 7}},
           qpoint[] = {3, 6, 5};
    int min_i = -1, min_d = 2147483647;

    for (int i = 0; i < sizeof(points)/sizeof(*points); ++i) {
        int dist = 0;

        for (int j = 0; j < sizeof(qpoint)/sizeof(*qpoint); ++j)
            dist += (points[i][j] - qpoint[j]) * (points[i][j] - qpoint[j]);

        if (min_d > dist) {
            min_d = dist;
            min_i = i;
        }
    }

    printf("%d\n", min_i);
    return 0;
}

File: nearest_neighbor/nn.go
package main
import "fmt"

func main() {
    points := [3][3]int{{1, 2, 3}, {3, 4, 5}, {5, 6, 7}}
    qpoint := [3]int{3, 6, 5}
    min_i  := -1
    min_d  := 2147483647

    for i := 0; i < len(points); i++ {
        dist := 0

        for j := 0; j < len(qpoint); j++ {
            dist += (points[i][j] - qpoint[j])*(points[i][j] - qpoint[j])
        }

        if min_d > dist {
            min_d = dist
            min_i = i
        }
    }

    fmt.Println(min_i)
}
